recipes/leveldb/all/conanfile.py

```diff
@@ -0,0 +1,71 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+
+class LevelDBCppConan(ConanFile):
+    name = "leveldb"
+    description = "LevelDB is a fast key-value storage library written at Google that provides an ordered mapping from string keys to string values."
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/google/leveldb"
+    topics = ("conan", "leveldb", "db")
+    license = "MIT"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False], 
+        "with_snappy": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_snappy": True,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_snappy:
+            self.requires("snappy/1.1.8")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["LEVELDB_BUILD_TESTS"] = False
+        self._cmake.definitions["LEVELDB_BUILD_BENCHMARKS"] = False
+        self._cmake.configure()
```

Comment:
@gocarlos below is a suggested change to avoid this issue.

This matters if you happen to use the recipe on a system that has a local libsnappy installed in which case, all the HAVE_SNAPPY sections in this file gets included in libleveldb: https://github.com/google/leveldb/blob/201f52201f5dd9701e7a8ceaa0ec4d344e69e022/port/port_stdcxx.h#L88-L99 causing the test_package build to fail linking:

```
[100%] Linking CXX executable bin/test_package
/usr/bin/ld: /home/jjkoshy/.conan/data/leveldb/1.22/_/_/package/b911f48570f9bb2902d9e83b2b9ebf9d376c8c56/lib/libleveldb.a(table_builder.cc.o): in function `leveldb::TableBuilder::WriteBlock(leveldb::BlockBuilder*, leveldb::BlockHandle*)':
table_builder.cc:(.text+0x629): undefined reference to `snappy::MaxCompressedLength(unsigned long)'
/usr/bin/ld: table_builder.cc:(.text+0x661): undefined reference to `snappy::RawCompress(char const*, unsigned long, char*, unsigned long*)'
/usr/bin/ld: /home/jjkoshy/.conan/data/leveldb/1.22/_/_/package/b911f48570f9bb2902d9e83b2b9ebf9d376c8c56/lib/libleveldb.a(format.cc.o): in function `leveldb::ReadBlock(leveldb::RandomAccessFile*, leveldb::ReadOptions const&, leveldb::BlockHandle const&, leveldb::BlockContents*)':
format.cc:(.text+0x48f): undefined reference to `snappy::GetUncompressedLength(char const*, unsigned long, unsigned long*)'
/usr/bin/ld: format.cc:(.text+0x4b3): undefined reference to `snappy::RawUncompress(char const*, unsigned long, char*)'
collect2: error: ld returned 1 exit status
```

In other words, the package's libleveldb depends on snappy regardless of the recipe's option to disable snappy.

The fix is simple - if have_snappy is set to false, then just patch CMakeLists.txt to remove the HAVE_SNAPPY portion.